pkg/generator.go

```diff
@@ -562,13 +562,22 @@ func (g *Generator) renderType(ctx context.Context, typ types.Type) string {
 	}
 }
 
-func (g *Generator) renderTypeTuple(ctx context.Context, tup *types.Tuple) string {
+func (g *Generator) renderTypeTuple(ctx context.Context, tup *types.Tuple, variadic bool) string {
 	var parts []string
 
 	for i := 0; i < tup.Len(); i++ {
 		v := tup.At(i)
 
-		parts = append(parts, g.renderType(ctx, v.Type()))
+		if variadic {
+			switch t := v.Type().(type) {
+			case *types.Slice:
+				parts = append(parts, "..."+g.renderType(ctx, t.Elem()))
+			default:
+				parts = append(parts, g.renderType(ctx, v.Type()))
```

Comment:
Is it possible for an argument to be variadic but not a slice?

Reply:
It looks like the variadic is true if one of args is variadic , so if we have multiple arguments, they will each pass in the for loop and we need to handle type

Reply:
See logs here

```
func (g *Generator) renderTypeTuple(ctx context.Context, tup *types.Tuple, variadic bool) string {
	var parts []string

	for i := 0; i < tup.Len(); i++ {
		v := tup.At(i)

		if variadic {
			log.Debug().Msgf("%+v", v.Type())
```

```
{"level":"debug","time":"2023-10-13T10:49:31+02:00","message":"string"}
{"level":"debug","time":"2023-10-13T10:49:31+02:00","message":"[]interface{}"}
{"level":"debug","time":"2023-10-13T10:49:31+02:00","message":"string"}
{"level":"debug","time":"2023-10-13T10:49:31+02:00","message":"[]interface{}"}
{"level":"debug","time":"2023-10-13T10:49:31+02:00","message":"string"}
{"level":"debug","time":"2023-10-13T10:49:31+02:00","message":"[]interface{}"}
```

Reply:
What happens if you have a function signature like this?

```go
func Foo(s []any, i ...int)
```
The `s` argument would match the `*types.Slice` case because it's  a slice, but it's not the variadic arg right?

Reply:
Yes good catch, I think we will have an error, but I think that Variadic information is not available for each arg
https://github.com/vektra/mockery/blob/e7c0bd979c493313b30aea4cf414da99be1bfcec/pkg/generator.go#L451
Any idea ?

Reply:
The variadic argument will always be the last one so you can probably just say 

```go
if variadic && i == tupl.Len()-1 {
```